controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -788,7 +789,9 @@ public CompletableFuture<CreateStreamStatus.Status> createStream(String scope, S
                                                                      long createTimestamp, long requestId) {
         log.debug(requestId, "createStream with resource called.");
         OperationContext context = streamMetadataStore.createStreamContext(scope, stream, requestId);
-        if (config.getRetentionPolicy() == null && this.globalRetentionPolicy.get()) {
+        if (!config.getRetentionPolicy().getRetentionType().equals(RetentionType.NONE)
+                && config.getRetentionPolicy() == null
+                && this.globalRetentionPolicy.get()) {
```

Comment:
Per our discussion, let's call this "default retention policy". I say this because it is currently misleading when it is set to SIZE. For example if one sets a "Global retention policy" of 500gb it sounds like all streams will be limited to 500gb collectively. However this is not the case, they are limited individually, and the total size may be much greater. Now that there is a explicit NONE which overrides it, the semantic is now: If no retention was specified it will apply this retention policy. Hence it is a default.

Reply:
Done